scalanativelib/test/src/mill/scalanativelib/HelloNativeWorldTests.scala

```diff
@@ -33,11 +39,10 @@ object HelloNativeWorldTests extends TestSuite {
       scalaNative <- Seq(scalaNative04, "0.4.9")
       mode <- List(ReleaseMode.Debug, ReleaseMode.ReleaseFast)
       if !(ZincWorkerUtil.isScala3(scala) && scalaNative == scalaNative04)
-    } yield (scala, scalaNative, mode)
+    } yield (scala, scalaNative, mode: ReleaseMode)
```

Comment:
Needing this type annotation is a limitation of type inference with the complicated `forSome` type we use in `Cross.Of`, where sub-types of the expected type are not properly widened. Providing an explicit annotation to give it the exact matching type seems to fix it